Lets Do This/Views/Reportback/LDTReportbackItemDetailView.m

```diff
@@ -129,4 +129,39 @@ - (IBAction)shareButtonTouchUpInside:(id)sender {
     }
 }
 
+- (void)setPreferredMaxLayoutWidthForLabels {
+    CGFloat width = CGRectGetWidth([UIScreen mainScreen].bounds);
+    NSLog(@"setPreferredMaxLayoutWidthForLabels width %f", width);
+    self.reportbackItemCaptionLabel.preferredMaxLayoutWidth = width - 16;
+    self.reportbackItemQuantityLabel.preferredMaxLayoutWidth = width / 3;
+    self.userCountryNameLabel.preferredMaxLayoutWidth = 100;
+}
+
+- (CGSize)preferredLayoutSizeFittingSize:(CGSize)targetSize {
+    CGRect originalFrame = self.frame;
+
+    // step1: set the detailView.frame to use our target width
+    self.frame = CGRectMake(self.frame.origin.x, self.frame.origin.y, targetSize.width, targetSize.height);
+    NSLog(@"targetSize.width %f", targetSize.width);
+
+    // step2: layout the cell
+    [self setNeedsLayout];
+    [self layoutIfNeeded];
+// Setting this to the desired width always results in the extra caption vertical padding. If we set it to just bounds.size.width like the SO thread suggests, we get vertical padding for first few rows but usually we eventually get the correct padding too.
+//    self.reportbackItemCaptionLabel.preferredMaxLayoutWidth = targetSize.width - 16;
+    self.reportbackItemCaptionLabel.preferredMaxLayoutWidth = self.reportbackItemCaptionLabel.bounds.size.width;
+    NSLog(@"reportbackItemCaptionLabel.preferredMaxLayoutWidth %f: ", self.reportbackItemCaptionLabel.preferredMaxLayoutWidth);
```

Comment:
Our originalFrame on a 6S width 295.000000, height 247.000000: 295 - 16 = 279.

Tried calling `[self setPreferredMaxLayoutWidthForLabels]` here instead of setting to the `bounds.size.width` -- which returns the expected 359 value yet still has incorrect heights. I've also tried setting the Share Button height and bottom constraint to 0 in the xib in hopes that would at least get things working and account for all of the extra padding. It's just not on every cell though... usually once you get towards the bottom things look correct.